File: 백준/Silver/13251. 조약돌 꺼내기/조약돌 꺼내기.js
const [m, input, k] = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");

const color = input.split(" ").map(Number);
const total = color.reduce((pre, cur) => pre + cur);

let ans = 0;
color.forEach((c) => {
  if (c >= k) {
    let value = 1;

    for (let i = 0; i < k; i++) {
      value *= (c - i) / (total - i);
    }

    ans += value;
  }
});

console.log(ans);

File: 백준/Silver/13251. 조약돌 꺼내기/조약돌 꺼내기.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int m = Integer.parseInt(br.readLine());

        String[] input = br.readLine().split(" ");

        int total = 0;
        int[] color = new int[m];
        for (int i = 0; i < m; i++) {
            color[i] = Integer.parseInt(input[i]);
            total += color[i];
        }

        int k = Integer.parseInt(br.readLine());

        if (m == 1 || k == 1) {
            System.out.println(1.0);
            return;
        }

        double mod = 1;
        for (int i = 0; i < k; i++) {
            mod *= (double) total - i;
        }

        double ans = 0;
        for (int i = 0; i < m; i++) {
            double temp = 1;
            for (int j = 0; j < k; j++) {
                temp *= (double) color[i] - j;
            }
            ans += temp / mod;
        }

        System.out.println(ans);
    }
}
